sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ChunkedDownloadUtils.java

```diff
@@ -100,6 +103,88 @@ && extractTotalBlobLength(blobStorageException.getResponse()
             });
     }
 
+    /*
+    Has a context value for additional download adjustments.
+
+    Download the first chunk. Construct a Mono which will emit the total count for calculating the number of chunks,
+    access conditions containing the etag to lock on, and the response from downloading the first chunk.
+     */
+    @SuppressWarnings("unchecked")
+    public static Mono<Tuple3<Long, BlobRequestConditions, BlobDownloadAsyncResponse>> downloadFirstChunk(
+        BlobRange range, ParallelTransferOptions parallelTransferOptions,
```

Comment:
Ah so in creating this new public method instead of forking + adding the adjusted blob length logic, I was thinking we would also update the [original version](https://github.com/Azure/azure-sdk-for-java/pull/41813/files#diff-e54b10e2121ed9d4e7ec0eacf29f1feca0ae11a1294ac2f6b9d56bee8897b47eL35) of this method to call into this new version, passing in a context of `null`. It should be similar to how we handle sharing logic across overloaded methods in the `EncryptedBlobAsyncClient` class. The main reason I'm suggesting this is so that we can avoid maintaining two versions of the code and instead just have the original version of the method call this new version directly.